pennylane/io/qasm_interpreter.py

```diff
@@ -94,14 +95,30 @@ def interpret(self, node: QASMNode, context: dict):
         context.update({"wires": [], "vars": {}, "gates": [], "callable": None})
 
         # begin recursive descent traversal
-        for value in node.__dict__.values():
-            if not isinstance(value, list):
-                value = [value]
-            for item in value:
-                if isinstance(item, QASMNode):
-                    self.visit(item, context)
+        try:
+            for value in node.__dict__.values():
+                if not isinstance(value, list):
+                    value = [value]
+                for item in value:
+                    if isinstance(item, QASMNode):
+                        self.visit(item, context)
+        except InterruptedError as e:
+            print(str(e))
```

Comment:
I would say we should instead just `return context` here. `end` is a perfectly valid command to encounter, so I don't think we particularly need to log it differently.  

Potentially this would be a case for logging if we wanted to add logging, but I don't think we should print anything out.